components/ledger/internal/services/command/update_balance_overdraft.go

```
@@ -0,0 +1,270 @@
+		UpdatedAt: now,
+	}
+
+	if _, cerr := uc.BalanceRepo.Create(ctx, overdraftBalance); cerr != nil {
+		// Benign race: a concurrent request won the Create. The partial
+		// UNIQUE index idx_unique_balance_account_key surfaces this as a
+		// PostgreSQL unique_violation (SQLSTATE 23505). Treat it as
+		// idempotent success — the peer request materialized the same row
+		// we were about to create. Verify with a follow-up Find so we
+		// never silently swallow a 23505 triggered by an unrelated index.
+		if isUniqueViolation(cerr) {
+			existing, reloadErr := uc.BalanceRepo.FindByAccountIDAndKey(ctx, organizationID, ledgerID, accountID, constant.OverdraftBalanceKey)
```

golangci-lint (2.11.4)

[error] shadow: declaration of "existing" shadows declaration at line 170

(govet)